dlt/destinations/weaviate/weaviate.py

```diff
@@ -0,0 +1,285 @@
+from types import TracebackType
+from typing import ClassVar, Optional, Sequence, Dict, Type, Iterable, Any
+import base64
+import binascii
+import zlib
+
+import weaviate
+
+from dlt.common import json, pendulum, logger
+from dlt.common.schema import Schema, TTableSchema, TSchemaTables
+from dlt.common.schema.typing import VERSION_TABLE_NAME, LOADS_TABLE_NAME
+from dlt.common.destination import DestinationCapabilitiesContext
+from dlt.common.destination.reference import (
+    NewLoadJob,
+    TLoadJobState,
+    LoadJob,
+    JobClientBase,
+)
+from dlt.common.storages import FileStorage
+from dlt.destinations.job_impl import EmptyLoadJob
+from dlt.destinations.job_client_impl import StorageSchemaInfo
+
+from dlt.destinations.weaviate import capabilities
+from dlt.destinations.weaviate.configuration import WeaviateClientConfiguration
+
+DLT_TABLE_PREFIX = "_dlt"
+
+
+# TODO: move to common
+def is_dlt_table(table_name: str) -> bool:
+    return table_name.startswith(DLT_TABLE_PREFIX)
+
+
+def snake_to_camel(snake_str: str) -> str:
+    return "".join(x.capitalize() for x in snake_str.split("_"))
+
+
+def table_name_to_class_name(table_name: str) -> str:
+    # Weaviate requires class names to be written with
+    # a capital letter first:
+    # https://weaviate.io/developers/weaviate/config-refs/schema#class
+    # For dlt tables strip the underscore from the name
+    # and make it all caps
+    # For non dlt tables make the class name camel case
+    return (
+        snake_to_camel(table_name)
+        if not is_dlt_table(table_name)
+        else table_name.lstrip("_").upper()
+    )
+
+
+class LoadWeaviateJob(LoadJob):
+    def __init__(
+        self,
+        table_name: str,
+        local_path: str,
+        db_client: weaviate.Client,
+        client_config: WeaviateClientConfiguration,
+        load_id: str,
+    ) -> None:
+        file_name = FileStorage.get_file_name_from_file_path(local_path)
+        super().__init__(file_name)
+
+        class_name = table_name_to_class_name(table_name)
+
+        with db_client.batch(
+            batch_size=client_config.weaviate_batch_size,
```

Comment:
I really hope that batch is all or nothing operation. otherwise we cannot guarantee that we do not duplicate something on retry... needs research